react/features/large-video/components/Labels.native.js

```diff
@@ -46,34 +141,173 @@ class Labels extends AbstractLabels<Props, *> {
+                        {
+                            this._renderTranscribingLabel()
+                        }
+                    </TouchableOpacity>
+                    {/*
+                      * Emil, Lyubomir, Nichole, and Zoli said that the Labels
+                      * should not be rendered in Picture-in-Picture. Saul
+                      * argued that the recording Labels should be rendered. As
+                      * a temporary compromise, don't render the
+                      * VideoQualityLabel at least because it's not that
+                      * important.
+                      */
+                        _reducedUI || (
+                            <TouchableOpacity
+                                onLayout = {
+                                    this._onLayout(LABEL_ID_QUALITY) }
+                                onPress = {
+                                    this._onPress(LABEL_ID_QUALITY) } >
+                                { this._renderVideoQualityLabel() }
+                            </TouchableOpacity>
+                        )
+                    }
+                </View>
                 {
-                    this._renderRecordingLabel(
-                        JitsiRecordingConstants.mode.STREAM)
-                }
-                {
-                    this._renderTranscribingLabel()
-                }
-                {/*
-                  * Emil, Lyubomir, Nichole, and Zoli said that the Labels
-                  * should not be rendered in Picture-in-Picture. Saul argued
-                  * that the recording Labels should be rendered. As a temporary
-                  * compromise, don't render the VideoQualityLabel at least
-                  * because it's not that important.
-                  */
-                    _reducedUI || this._renderVideoQualityLabel()
+                    this._renderExpandedLabel()
                 }
             </View>
         );
     }
 
+    _onLayout: string => Function
+
+    /**
+     * Creates a function to be invoked when the onLayout of the touchables are
+     * triggered.
+     *
+     * @param {string} label - The identifier of the label that's onLayout is
+     * triggered.
+     * @returns {Function}
+     */
+    _onLayout(label) {
+        return ({ nativeEvent: { layout } }) => {
+            const { labelLayouts } = this.state;
+            const updatedLayout = {};
+
+            updatedLayout[label] = layout;
+            this.setState({
+                labelLayouts: {
+                    ...labelLayouts,
+                    ...updatedLayout
+                }
+            });
+        };
+    }
+
+    _onPress: string => Function
+
+    /**
+     * Creates a function to be invoked when the onPress of the touchables are
+     * triggered.
+     *
+     * @param {string} label - The identifier of the label that's onLayout is
+     * triggered.
+     * @returns {Function}
+     */
+    _onPress(label) {
```

Comment:
The name suggests a handler not a creator function. Can you try to find a better name?